transform/instance.go

```diff
@@ -367,6 +367,75 @@ func (ot *objectTransformer) transform(in interface{}, modifier pathModifier) (i
 
 }
 
+// transform retrieves the value for this object by building the value for the base object and then adding in any
+// transforms for all defined child fields. If a transform is provided it transforms the children relative to the
+// passed in node
+func (ot *objectTransformer) objectTransformXML(in interface{}, modifier pathModifier) (interface{}, error) {
+	path := ot.jsonPath
+	if modifier != nil {
+		path = modifier(path)
+	}
+	var newValue map[string]interface{}
+
+	// For the object use a transform if it exists or the default or an empty map
+	if ot.transforms != nil {
+		rawValue, err := ot.transforms.transform(in, "object", modifier, ot.format)
+		if err != nil {
+			return nil, err
+		}
+
+		switch v := rawValue.(type) {
+		case *xmlquery.Node:
+			in = v
+		case []*xmlquery.Node:
+			if len(v) > 0 {
+				in = v[0]
+			}
+		default:
+			return nil, errors.New("non xml node returned from object transform")
+		}
+	}
+
+	if newValue == nil {
```

Comment:
Looks like newValue is always nil since it was just defined up above. I would drop the if statement and consider bring the variable definition here since it isn't used until this point.